File: src/domain/schemas/match.ts
import { z } from "zod";
import { candidateSchema } from "./candidate";
import { opportunitySchema } from "./opportunity";

export const matchSchema = z.object({
  id: z.string(),
  opportunity: opportunitySchema,
  candidate: candidateSchema,
  submissionId: z.string(),
  createdAt: z.string().datetime(),
  status: z.enum(["matched", "interview_requested", "interview_scheduled", "closed"]),
});

export type Match = z.infer<typeof matchSchema>;

File: backend/app/schemas/match.py
from datetime import datetime
from enum import StrEnum

from pydantic import BaseModel

from app.schemas.opportunity import OpportunityResponse
from app.schemas.reaction import EmployerReactionResponse
from app.schemas.submission import CandidateResponse


class MatchStatus(StrEnum):
    matched = "matched"
    interview_requested = "interview_requested"
    interview_scheduled = "interview_scheduled"
    closed = "closed"


class MatchResponse(BaseModel):
    id: str
    opportunity: OpportunityResponse
    candidate: CandidateResponse
    submissionId: str
    createdAt: datetime
    status: MatchStatus


class EmployerSubmissionReactionResponse(BaseModel):
    reaction: EmployerReactionResponse
    match: MatchResponse | None = None
